### PR TITLE
Ordering by `fformatter` field.

### LINKED ISSUE #806: Sort formatted values in vector API response (`ordering` + `fformatter` params)
### Checklist

- [X] I've searched through the [current issues](https://github.com/g3w-suite/g3w-admin/issues) to make sure this feature hasn't been requested already.

### Motivation

Be able to sort the the following values by "formatted" field value instead of by field key:

https://v37.g3wsuite.it/vector/api/data/qdjango/97/buildings_2f43dc1d_6725_42d2_a09b_dd446220104a/?fformatter=address&ordering=address

```jsonc
// current behavior

"data": [
  [ "A101", "Via Sallustio Bandini" ],
  [ "A102", "Via Atto Vannucci" ],
  [ "A103", "Via Alfredo Oriani gia' Via Vecchia" ],
  [ "A106", "Piazza Pietro Leopoldo" ],
  [ "A107", "Via Ubaldo Montelatici" ],
  [ "A108", "Via Giandomenico Romagnosi" ],
  [ "B002", "Via Giovanni Lampredi" ],
  [ "C106", "Via Pietro Cironi" ],
  [ "C107", "Via Pietro Corridoni" ]
],
```

### Suggested solution

Introduce a new parameter or change the `ordering` behavior so that response can be like following:

```jsonc
// desired behavior

"data": [
  [ "A106", "Piazza Pietro Leopoldo" ],
  [ "A103", "Via Alfredo Oriani gia' Via Vecchia" ],
  [ "A102", "Via Atto Vannucci" ],
  [ "A108", "Via Giandomenico Romagnosi" ],
  [ "B002", "Via Giovanni Lampredi" ],
  [ "C106", "Via Pietro Cironi" ],
  [ "C107", "Via Pietro Corridoni" ],
  [ "A101", "Via Sallustio Bandini" ],
  [ "A107", "Via Ubaldo Montelatici" ]
],
```

**NB** the sorting algo must also take into account **numeric**, **null** and **boolean** values (see below)

### Alternatives considered

Formatted values can be sorted on client-side, but this make every API calls code even more complex/unreliable:

```js
fetch('https://v37.g3wsuite.it/vector/api/data/qdjango/97/buildings_2f43dc1d_6725_42d2_a09b_dd446220104a/?fformatter=address' /* + '&ordering=address' */)
  .then(d => d.json())
  // sorted by fformatter
  .then(d => d.data.sort((a, b) => `${a[1]}`.localeCompare(b[1], undefined, 'number' === typeof a[1] ? { numeric: true } : { sensitivity: 'base' })))
```

### PR DESCRIPTION
**Pull request by wlorenzetti**
Closes: #806 

Apply the orderign URl parameter on field set by `fformatter` parameter.

Without `ordering` parameter:
```
http://localhost:8006/vector/api/data/qdjango/336/buildings_2f43dc1d_6725_42d2_a09b_dd446220104a/?fformatter=address&format=json
```

```json
  "data": [
    [ "A101",  "Via Sallustio Bandini" ],
    [ "A102",   "Via Atto Vannucci" ],
    [ "A103",   "Via Alfredo Oriani gia' Via Vecchia" ],
    [ "A106",   "Piazza Pietro Leopoldo" ],
    [ "A107",   "Via Ubaldo Montelatici" ],
    [ "A108",   "Via Giandomenico Romagnosi" ],
    [ "B002",  "Via Giovanni Lampredi" ]
]
```

With `ordering` parameter (_ordering=address_):

http://localhost:8006/vector/api/data/qdjango/336/buildings_2f43dc1d_6725_42d2_a09b_dd446220104a/?fformatter=address&format=json&ordering=address

```json
"data": [
    [ "A106",  "Piazza Pietro Leopoldo" ],
    [ "A103",  "Via Alfredo Oriani gia' Via Vecchia" ],
    [ "A102",  "Via Atto Vannucci" ],
    [ "A108",  "Via Giandomenico Romagnosi" ],
    [ "B002",  "Via Giovanni Lampredi" ],
    [ "A101",  "Via Sallustio Bandini" ],
    [ "A107",  "Via Ubaldo Montelatici" ]
  ]
```

Also works with `-` i.e. `ordering=-address` fro reversing results.

### IMPORTANT
The sorting of results is made by `str.sort()` python function

### CODE REVIEW

File: g3w-admin/core/api/base/views.py [hunk 1]
```diff
@@ -586,7 +590,13 @@ def response_data_mode(self, request, export_features=False):
...
 
             # sort values
-            values.sort()
+            if ('fformatter' in self.request_data
+                and 'ordering' in self.request_data
+                and self.request_data['fformatter'] in self.request_data['ordering']):
+                rev = True if self.request_data['ordering'].startswith('-') else False
+                values.sort(reverse=rev, key=lambda e: e[1])
+            else:
+                values.sort()
```
Comment (Raruto) [inline g3w-admin/core/api/base/views.py]: > The sorting of results is made by `str.sort()` python function

@wlorenzetti does it work with mixed arrays?

image

Comment (wlorenzetti) [reply g3w-admin/core/api/base/views.py]: The result for this API is  list of types that are identical to each other

Comment (Raruto) [reply g3w-admin/core/api/base/views.py]: Ok, the case above is purposely unlikely (with all different types).

But the real question is, it should also handle null values?

For example:

```
"data": [
    [ "A106",  "Piazza Pietro Leopoldo" ],
    [ "A103",  None],
    [ "A102",  "Via Atto Vannucci" ],
    [ "A108",  "Via Giandomenico Romagnosi" ],
    [ "B002",  "Via Giovanni Lampredi" ],
    [ "A101",  None],
    [ "A107",  "Via Ubaldo Montelatici" ]
  ]
```

Also for the previous example the error was generated on the `NoneType` value.

Comment (wlorenzetti) [reply g3w-admin/core/api/base/views.py]: Ok I just make same tests.

**Review by Raruto — APPROVED** [review]

**Pull request merged**